test/Microsoft.Extensions.Internal.Test/WebEncodersTests.cs

```diff
@@ -109,5 +176,179 @@ public void Base64UrlEncode_BadOffsets(int inputLength, int offset, int count)
                 var retVal = WebEncoders.Base64UrlEncode(input, offset, count);
             });
         }
+
+        [Theory]
+        [InlineData(0, 0)]
+        [InlineData(2, 4)]
+        [InlineData(3, 4)]
+        [InlineData(4, 4)]
+        [InlineData(6, 8)]
+        [InlineData(7, 8)]
+        public void GetArraySizeRequiredToDecode(int inputLength, int expectedPadding)
+        {
+            var result = WebEncoders.GetArraySizeRequiredToDecode(inputLength);
+
+            Assert.Equal(expectedPadding, result);
+        }
+
+        [Fact]
+        public void GetArraySizeRequiredToDecode_NegativeInputLength_Throws()
+        {
+            var exception = Assert.Throws<ArgumentOutOfRangeException>(() => WebEncoders.GetArraySizeRequiredToDecode(-1));
+            Assert.Equal("count", exception.ParamName);
+        }
+
+        [Theory]
+        [InlineData(1)]
+        [InlineData(5)]
+        public void GetArraySizeRequiredToDecode_MalformedInputLength(int inputLength)
+        {
+            Assert.Throws<FormatException>(() =>
+            {
+                var retVal = WebEncoders.GetArraySizeRequiredToDecode(inputLength);
+            });
+        }
+
+        // Taken from https://github.com/aspnet/HttpAbstractions/pull/926
+        [Theory]
+        [InlineData("_", "/")]
+        [InlineData("-", "+")]
+        [InlineData("a-b-c", "a+b+c=")]
+        [InlineData("a_b_c_d", "a/b/c/d=")]
+        [InlineData("a-b_c", "a+b/c==")]
+        [InlineData("a-b_c-d", "a+b/c+d=")]
+        [InlineData("abcd", "abcd")]
+        public void UrlDecode_ReturnsValid_Base64String(string text, string expectedValue)
```

Comment:
I needed a safety net for the private class `UrlEncoder`, so these tests were added. 
Is it OK to test implementation details or shall I remove these tests?